cmd/dep/ensure.go

```diff
@@ -127,6 +127,9 @@ func (cmd *ensureCommand) Run(ctx *dep.Ctx, args []string) error {
 	if err != nil {
 		return errors.Wrap(err, "ensure ListPackage for project")
 	}
+	if containsOnlyErrorsOrIsEmpty(params.RootPackageTree.Packages) {
+		return errors.New("Root package doesn't contain any valid go package")
```

Comment:
I think it would be preferable to provide different error text for two, possibly three cases:

1. There's go code, but it has errors
2. There's no go code in the tree at all
3. (possibly) There's no tree, just an empty current working directory